src/mongocxx/collection.cpp

```diff
@@ -125,6 +124,10 @@ cursor collection::find(bsoncxx::document::view filter, const options::find& opt
         filter_builder << "$orderby" << bsoncxx::types::b_document{*options.sort()};
     }
 
+    if (options.hint()) {
+        filter_builder << builder::stream::concatenate{options.hint()->to_document()};
```

Comment:
This pattern that we use throughout the codebase is bad, because to_document() returns a temporary value that concatenate{} converts to a view and drops on the floor.  View_or_value will fix it!